python/lsst/daf/butler/core/config.py

```diff
@@ -540,16 +540,15 @@ def _splitIntoKeys(key):
             hierarchy = key.split(d)
             if temp:
                 hierarchy = [h.replace(temp, d) for h in hierarchy]
-            return hierarchy
+            # Copy the list to keep mypy quiet.
+            return list(hierarchy)
         elif isinstance(key, Iterable):
-            return list(key)
+            return [k for k in key]
         else:
-            # Not sure what this is so try it anyway
-            return [
-                key,
-            ]
+            # Do not try to guess.
+            raise TypeError(f"Provided key [{key}] neither str nor iterable.")
```

Comment:
I'm assuming this is okay. mypy didn't like the fallback code and so it was easier for me to change it to a raise. If jenkins passes it's likely okay.